PathPlanner/RobotNode.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2016 PolySync
+ *
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <cmath>
+#include <iostream>
+
+#include <PolySyncDataModel.hpp>
+
+#include "RobotNode.hpp"
+
+using namespace cv;
+using namespace std;
+using namespace polysync::datamodel;
+
+constexpr int INVALID_LOC = -1;
+
+/**
+ * \example RobotNode.cpp
+ *
+ * PolySync Path Planner Example.
+ *
+ * Shows how to subscribe a simulated robot node to a planning algorithm node.
+ * Robot node listens for platform motion messages with desired waypoints from
```

Comment:
May be worth noting how the platform motion message is being used in this context. 

Ideally this would create and use a waypoints message.